
# default controller
optimus_joint_state_controller:
  type: "joint_state_controller/JointStateController"
  publish_rate: 50

optimus_FR_outer_wheel_controller:
    type: velocity_controllers/JointVelocityController
    joint: 'FR_outer_wheel_joint'
    pid: {p: 10.0, i: 0.0, d: 0.0}

optimus_FL_outer_wheel_controller:
    type: velocity_controllers/JointVelocityController
    joint: 'FL_outer_wheel_joint'
    pid: {p: 10.0, i: 0.0, d: 0.0}

optimus_FR_inner_wheel_controller:
    type: velocity_controllers/JointVelocityController
    joint: 'FR_inner_wheel_joint'
    pid: {p: 10.0, i: 0.0, d: 0.0}

optimus_FL_inner_wheel_controller:
    type: velocity_controllers/JointVelocityController
    joint: 'FL_inner_wheel_joint'
    pid: {p: 10.0, i: 0.0, d: 0.0}

optimus_FR_outer_wheel_connector_controller:
    type: position_controllers/JointPositionController
    joint: 'FR_outer_wheel_connector_joint'
    pid: {p: 0.1, i: 0.0, d: 0.0}

optimus_FL_outer_wheel_connector_controller:
    type: position_controllers/JointPositionController
    joint: 'FL_outer_wheel_connector_joint'
    pid: {p: 0.1, i: 0.0, d: 0.0}

optimus_FR_inner_wheel_connector_controller:
    type: position_controllers/JointPositionController
    joint: 'FR_inner_wheel_connector_joint'
    pid: {p: 0.1, i: 0.0, d: 0.0}

optimus_FL_inner_wheel_connector_controller:
    type: position_controllers/JointPositionController
    joint: 'FL_inner_wheel_connector_joint'
    pid: {p: 0.1, i: 0.0, d: 0.0}

optimus_R_docker_shaft_controller:
    type: position_controllers/JointPositionController
    joint: 'R_docker_shaft_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

optimus_L_docker_shaft_controller:
    type: position_controllers/JointPositionController
    joint: 'L_docker_shaft_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

optimus_RR_outer_wheel_controller:
    type: velocity_controllers/JointVelocityController
    joint: 'RR_outer_wheel_joint'
    pid: {p: 10.0, i: 0.0, d: 0.0}

optimus_RL_outer_wheel_controller:
    type: velocity_controllers/JointVelocityController
    joint: 'RL_outer_wheel_joint'
    pid: {p: 10.0, i: 0.0, d: 0.0}

optimus_RR_inner_wheel_controller:
    type: velocity_controllers/JointVelocityController
    joint: 'RR_inner_wheel_joint'
    pid: {p: 10.0, i: 0.0, d: 0.0}

optimus_RL_inner_wheel_controller:
    type: velocity_controllers/JointVelocityController
    joint: 'RL_inner_wheel_joint'
    pid: {p: 10.0, i: 0.0, d: 0.0}

optimus_RR_outer_wheel_connector_controller:
    type: position_controllers/JointPositionController
    joint: 'RR_outer_wheel_connector_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

optimus_RL_outer_wheel_connector_controller:
    type: position_controllers/JointPositionController
    joint: 'RL_outer_wheel_connector_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

optimus_RR_inner_wheel_connector_controller:
    type: position_controllers/JointPositionController
    joint: 'RR_inner_wheel_connector_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

optimus_RL_inner_wheel_connector_controller:
    type: position_controllers/JointPositionController
    joint: 'RL_inner_wheel_connector_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

# UR10 Front Control
UR10_front_shoulder_pan_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_front_shoulder_pan_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

UR10_front_shoulder_lift_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_front_shoulder_lift_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

UR10_front_elbow_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_front_elbow_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

UR10_front_wrist_1_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_front_wrist_1_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

UR10_front_wrist_2_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_front_wrist_2_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

UR10_front_wrist_3_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_front_wrist_3_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

# UR10 Rear Control
UR10_rear_shoulder_pan_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_rear_shoulder_pan_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

UR10_rear_shoulder_lift_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_rear_shoulder_lift_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

UR10_rear_elbow_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_rear_elbow_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

UR10_rear_wrist_1_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_rear_wrist_1_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

UR10_rear_wrist_2_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_rear_wrist_2_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}

UR10_rear_wrist_3_joint_controller:
    type: position_controllers/JointPositionController
    joint: 'ur10e_rear_wrist_3_joint'
    pid: {p: 1.0, i: 0.0, d: 0.0}